src/ck-ldb/lb_strategy.h

```diff
@@ -0,0 +1,235 @@
+#ifndef LB_STRATEGY_H
+#define LB_STRATEGY_H
+
+#include <random>
+#include <algorithm>
+
+namespace lb_strategy
+{
+
+  template <typename T, bool is_ptr = std::is_pointer<T>::value> struct CmpLoadGreater {};
+
+  template <typename T> struct CmpLoadGreater<T,true> {
+    inline bool operator()(const T a, const T b) const { return (a->getLoad() > b->getLoad()); }
+  };
+
+  template <typename T> struct CmpLoadGreater<T,false> {
+    inline bool operator()(const T &a, const T &b) const { return (a.getLoad() > b.getLoad()); }
+  };
+
+  template <typename T, bool is_ptr = std::is_pointer<T>::value> struct CmpLoadLess {};
+
+  template <typename T> struct CmpLoadLess<T,true> {
+    inline bool operator()(const T a, const T b) const { return (a->getLoad() < b->getLoad()); }
+  };
+
+  template <typename T> struct CmpLoadLess<T,false> {
+    inline bool operator()(const T &a, const T &b) const { return (a.getLoad() < b.getLoad()); }
+  };
+
+  template <typename T, bool is_ptr = std::is_pointer<T>::value> struct CmpId {};
+
+  template <typename T> struct CmpId<T,true> {
+    inline bool operator()(const T a, const T b) const { return (a->id < b->id); }
+  };
+
+  template <typename T> struct CmpId<T,false> {
+    inline bool operator()(const T &a, const T &b) const { return (a.id < b.id); }
+  };
+
+  template <typename T>
+  T *ptr(T &obj) { return &obj; }  // turn reference into pointer
+
+  template <typename T>
+  T *ptr(T *obj) { return obj; }  // obj is already pointer, return it
+
+  // ---------------- Obj --------------------
+
+  struct obj_1_data { float load; };
+  template <int N> struct obj_N_data { float load[N]; float maxload; };
+
+  template <int N, bool multi = (N > 1)>
+  class Obj : public std::conditional<multi, obj_N_data<N>, obj_1_data>::type
+  {
+  public:
+
+    unsigned int id;
```

Comment:
Nitpick: `unsigned` is often not a good idea. Is there a reason why `id` is unsigned here (in contrast to Proc's id, for example)?